src/vulkan/vulkan_initializers.cpp

```diff
@@ -188,38 +187,37 @@ namespace IC {
-                                                      std::vector<std::shared_ptr<PointLight>> &pointLights,
-                                                      glm::mat4 viewMat) {
+    SceneLightDescriptors CreateSceneLightDescriptors(SceneLightData &lightData, glm::mat4 viewMat) {
         SceneLightDescriptors descriptors;
-        glm::vec3 directionalViewSpaceDirection = viewMat * glm::vec4(directionalLight->direction, 0.0f);
+        glm::vec3 directionalViewSpaceDirection = viewMat * glm::vec4(lightData.directionalLight->Direction(), 0.0f);
 
         DirectionalLightDescriptors directionalDescriptors{};
         directionalDescriptors.dir = directionalViewSpaceDirection;
-        directionalDescriptors.diff = directionalLight->color;
-        directionalDescriptors.amb = directionalLight->ambient;
-        directionalDescriptors.spec = directionalLight->specular;
+        directionalDescriptors.diff = lightData.directionalLight->Color();
+        directionalDescriptors.amb = lightData.directionalLight->Ambient();
+        directionalDescriptors.spec = lightData.directionalLight->Specular();
         descriptors.directionalLight = directionalDescriptors;
-        for (int i = 0; i < pointLights.size() && i < MAX_POINT_LIGHTS; i++) {
-            glm::vec3 lightViewSpacePos = viewMat * glm::vec4(pointLights[i]->previewMesh.pos, 1.0f);
+
+        for (int i = 0; i < lightData.pointLights.size() && i < MAX_POINT_LIGHTS; i++) {
+            glm::vec3 lightViewSpacePos = viewMat * glm::vec4(lightData.pointLights[i].transform->Position(), 1.0f);
 
             PointLightDescriptors pointLightDescriptors{};
             pointLightDescriptors.pos = lightViewSpacePos;
-            pointLightDescriptors.amb = pointLights[i]->ambient;
-            pointLightDescriptors.diff = pointLights[i]->color;
-            pointLightDescriptors.spec = pointLights[i]->specular;
-            pointLightDescriptors.cons = pointLights[i]->constant;
-            pointLightDescriptors.lin = pointLights[i]->linear;
-            pointLightDescriptors.quad = pointLights[i]->quadratic;
+            pointLightDescriptors.amb = lightData.pointLights[i].light->Ambient();
+            pointLightDescriptors.diff = lightData.pointLights[i].light->Color();
+            pointLightDescriptors.spec = lightData.pointLights[i].light->Specular();
+            pointLightDescriptors.cons = lightData.pointLights[i].light->Constant();
+            pointLightDescriptors.lin = lightData.pointLights[i].light->Linear();
+            pointLightDescriptors.quad = lightData.pointLights[i].light->Quadratic();
 
             descriptors.pointLights[i] = pointLightDescriptors;
         }
-        descriptors.numPointLights = pointLights.size() > MAX_POINT_LIGHTS ? MAX_POINT_LIGHTS : pointLights.size();
+        descriptors.numPointLights =
+            lightData.pointLights.size() > MAX_POINT_LIGHTS ? MAX_POINT_LIGHTS : lightData.pointLights.size();
```

Comment:
You can actually do this much cleaner with `std::min(lightData.pointLights.size(), MAX_POINT_LIGHTS)`, think you might need to `#include <functional>` though